set.mm

```diff
@@ -29488,41 +29488,238 @@ atomic formula (class version of ~ elsb1 ).  Usage of this theorem is
                    -> ( A. x e. A A. y e. B ps -> A. x e. A A. y e. B ch ) ) $=
       ( wral cv wcel wa wi anassrs ralimdva ) ABEGICEGIDFADJFKZLBCEGAPEJGKBCMHN
       OO $.
+
+    $( Deduction doubly quantifying both antecedent and consequent, based on
+       Theorem 19.22 of [Margaris] p. 90.  (Contributed by AV, 5-Jan-2022.) $)
+    reximdvva $p |- ( ph
+                   -> ( E. x e. A E. y e. B ps -> E. x e. A E. y e. B ch ) ) $=
+      ( wrex cv wcel wa wi anassrs reximdva ) ABEGICEGIDFADJFKZLBCEGAPEJGKBCMHN
+      OO $.
+  $}
+
+  ${
+    $d x y ph $.  $d x y ps $.  $d y A $.
+    ralrimdvva.1 $e |- ( ( ph /\ ( x e. A /\ y e. B ) ) -> ( ps -> ch ) ) $.
+    $( Inference from Theorem 19.21 of [Margaris] p. 90.  (Restricted
+       quantifier version with double quantification.)  (Contributed by NM,
+       2-Feb-2008.) $)
+    ralrimdvva $p |- ( ph -> ( ps -> A. x e. A A. y e. B ch ) ) $=
+      ( cv wcel wa wi ex com23 ralrimdvv ) ABCDEFGADIFJEIGJKZBCAPBCLHMNO $.
+  $}
```

Comment:
ralrimdvva et al seems 2 dimensional while rgen3 above is 3, unless I'm misunderstanding dimensions, or there is no preference for second vs third dimension

similar with 3reeanv and 2ralor

Reply:
So far, I only distinguish between one- and multi-dimensional, keeping theorems similar in structure like rgen2 and rgen3 still close together.  If possible, lets first establish an overall rewrite, before getting into details.  This fine tuning need not be done by me, by the way.